lib/generator.rb

```diff
@@ -36,12 +37,24 @@ def test_cases
   end
 
   def generate
+    generate_test_file
+    update_version
+  end
+
+  private
+
+  def generate_test_file
     File.open(path_to("#{name.gsub(/[ -]/, '_')}_test.rb"), 'w') do |f|
       f.write ERB.new(File.read(path_to('example.tt'))).result binding
+      FileUtils.chmod('+x', f)
```

Comment:
The nice thing about this is that you will often generate the file a few times while working iteratively toward a solution. This makes it so that you don't even have to worry about the file mode each time you generate the test file.